explorer/interpreter/impl_scope.h

```diff
@@ -176,7 +176,7 @@ class ImplScope {
                                const TypeChecker& type_checker) const
       -> ErrorOr<std::optional<ResolveResult>>;
 
-  std::vector<Impl> impls_;
+  std::vector<Impl> impl_declarations_;
```

Comment:
Hm, these are more general than impl declarations; this also includes things like the "`T` implements `Hashable`" fact we get from `fn F[T:! Hashable](x: T)` or from `fn F[T:! type where T impls Hashable](x: T)`.

I don't have a great name for these in mind, but I think `impl_declarations` is probably too confusing. The least-worst idea I have so far is to rename class `Impl` to `ImplFact` and call this `impl_facts`. But I don't like it!